source/Octopus.Data/Model/User/ActiveDirectoryIdentity.cs

```diff
@@ -0,0 +1,17 @@
+namespace Octopus.Data.Model.User
+{
+    public class ActiveDirectoryIdentity : ExternalIdentity
```

Comment:
This is the one thing that smells to me. It's nice to have the strong typing for the multiple keys we use for, say, an `ActiveDirectoryIdentity` - but it feels like either:

- this class is defined in the wrong place and should be part of the Directory Services provider itself; or
- the class should be made more generic where you can store/retrieve external ids by some kind of key